src/c2goto/library/kernel.c

```diff
@@ -107,3 +109,38 @@ unsigned long copy_from_user(void *to, void *from, unsigned long size)
 
   return 0;
 }
+
+void spin_lock_init(spinlock_t *lock)
+{
+  //check if the lock is valid
+  assert(lock != NULL);
+  //initialize the lock
+  lock->locked = false;
+}
+
+bool spin_lock(spinlock_t *lock)
```

Comment:
This is related to your answer above:
> if the limit is reached, then it means the acquiring of the lock failed.

According to my understand of the Linux kernel's sources, `spin_lock()` can never fail, see, e.g., include/linux/spinlock_rt.h:
```
static __always_inline void spin_lock(spinlock_t *lock)
{
        rt_spin_lock(lock);
}
```
There is a `trylock` function that may fail. Maybe you are modelling that one?